### PR TITLE
Add peer registration to the secure overlay network

### PR DESCRIPTION
**Pull request by AndrewSC208**
Implement `RegisterPeer` rpc method on the server. This allows a client to generate credentials for a peer to register to the overlay network.

### CODE REVIEW

File: services/platform/server/apps/controller.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ type (
 func NewController(logger chassis.Logger) Controller {
 	chassis.GetConfig().SetDefault(autoUpdateCronConfigKey, "0 3 * * *")
 	return &controller{
-		k8sclient: k8sclient.NewClient(logger),
+		k8sclient: nil,
```
Comment (jgkawell) [inline services/platform/server/apps/controller.go:56]: Need to add this back

File: services/platform/server/main.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func main() {
 		sctl        = system.NewController(logger, broadcaster)
 		lctl        = locator.NewController(logger)
 		webRPC      = web.New(logger, actl, sctl, lctl)
-		webHTTP     = web.NewHttp(logger, actl, sctl)
+		// webHTTP     = web.NewHttp(logger, actl, sctl)
```
Comment (jgkawell) [inline services/platform/server/main.go:31]: need to add this back too

File: services/platform/server/apps/controller.go (hunk 1 above)
Comment (AndrewSC208) [reply services/platform/server/apps/controller.go:56]: ohhh crap

File: services/platform/server/system/controller.go [hunk 3]
```diff
@@ -51,7 +52,7 @@ func NewController(logger chassis.Logger, broadcaster async.Broadcaster) Control
 	config.SetDefault(osAutoUpdateCronConfigKey, "0 1 * * *")
 	config.SetDefault(containerAutoUpdateCronConfigKey, "0 2 * * *")
 	return &controller{
-		k8sclient:        k8sclient.NewClient(logger),
+		k8sclient:        nil,
```
Comment (jgkawell) [inline services/platform/server/system/controller.go:55]: revert this

File: services/platform/server/apps/controller.go (hunk 1 above)
Comment (AndrewSC208) [reply services/platform/server/apps/controller.go:56]: TY for seeing that

Comment (AndrewSC208) [reply services/platform/server/apps/controller.go]: updated

File: services/platform/server/system/controller.go (hunk 3 above)
Comment (AndrewSC208) [reply services/platform/server/system/controller.go]: updated

File: services/platform/server/main.go (hunk 2 above)
Comment (AndrewSC208) [reply services/platform/server/main.go]: udpated

File: services/platform/server/system/peering.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		return nil, err
+	}
+
+	err = com.Send(&dv1.ServerMessage{
+		Message: &dv1.ServerMessage_AddWireguardPeer{
+			AddWireguardPeer: &dv1.AddWireguardPeer{
+				Peer: &dv1.WireguardPeer{
+					PublicKey: pubKey.String(),
+					// The assumption is that any device using wireguard in the network can talk to each other
+					AllowedIps: []string{"*", "0.0.0.0"},
```
Comment (jgkawell) [inline services/platform/server/system/peering.go:81]: I think we'll want to revisit this later on but this is fine for now

**Review by jgkawell — APPROVED** [review]

**Pull request merged**